### ISSUE TITLE
[translator] Pimple 3.2.0 breaks translator

### ISSUE THREAD

**Issue by quazardous** [silexphp/Silex#1540]
```
PHP Fatal error:  Uncaught Pimple\Exception\InvalidServiceIdentifierException: Identifier "translator.resources" does not contain an object definition. in /home/david/www/m2m/vendor/pimple/pimple/src/Pimple/Container.php:245
Stack trace:
#0 /home/david/www/m2m/src/M2m/Provider/TranslationServiceProvider.php(65): Pimple\Container->extend('translator.reso...', Object(Closure))
#1 /home/david/www/m2m/vendor/pimple/pimple/src/Pimple/Container.php(286): M2m\Provider\TranslationServiceProvider->register(Object(M2m\Application))
#2 /home/david/www/m2m/vendor/silex/silex/src/Silex/Application.php(90): Pimple\Container->register(Object(M2m\Provider\TranslationServiceProvider), Array)
#3 /home/david/www/m2m/app/bootstrap.php(74): Silex\Application->register(Object(M2m\Provider\TranslationServiceProvider))
#4 /home/david/www/m2m/cli-config.php(4): include('/home/david/www...')
#5 /home/david/www/m2m/vendor/doctrine/orm/bin/doctrine.php(55): require('/home/david/www...')
#6 /home/david/www/m2m/vendor/doctrine/orm/bin/doctrine(4): include( in /home/david/www/m2m/vendor/pimple/pimple/src/Pimple/Container.php on line 245

```
the [doc](https://github.com/silexphp/Silex/blob/master/doc/providers/translation.rst) says to do

```
$app->extend('translator.resources', function ($resources, $app) {
    $resources = array_merge($resources, array(
        array('array', array('This value should be a valid number.' => 'Cette valeur doit être un nombre.'), 'fr', 'validators'),
    ));

    return $resources;
});
```

but it does not work anymore.

the obvious fix is to override 

```
        $app['translator.resources'] = $app->protect(function ($app) {
            return array();
        });
```

but it's not very friendly nor easily sharable between providers.

**Comment by SpacePossum** [comment]
maybe related to https://github.com/silexphp/Silex/pull/1538?

**Comment by quazardous** [comment]
yes sorry... should I close ?

**Comment by skalpa** [comment]
I guess you can, there is already a fix ready for Pimple too in silexphp/Pimple#239 that replaces the exception by a deprecation notice to ensure user providers don't just break.
